support iec flavors data source and docs

Lance52259 opened the pull request:
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccIECFlavorsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccIECFlavorsDataSource_basic
=== PAUSE TestAccIECFlavorsDataSource_basic
=== CONT  TestAccIECFlavorsDataSource_basic
--- PASS: TestAccIECFlavorsDataSource_basic (14.13s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       14.177s

Lance52259 committed:
support iec flavors data source and docs

Lance52259 committed:
Add iec_flavors.md to data source docs

ShiChangkuo commented on huaweicloud/data_source_huaweicloud_iec_flavors_test.go:
```diff
@@ -0,0 +1,53 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIECFlavorsDataSourceID(resourceName, rName),
+					resource.TestMatchResourceAttr(resourceName, "flavors.#", regexp.MustCompile("[1-9]\\d*")),
+					resource.TestCheckResourceAttr(resourceName, "region", "cn-north-4"),
```
please use the "HW_REGION_NAME" env instead of a specified region

ShiChangkuo commented on huaweicloud/data_source_huaweicloud_iec_flavors_test.go:
```diff
@@ -0,0 +1,53 @@
+	})
+}
+
+func testAccCheckIECFlavorsDataSourceID(n, rName string) resource.TestCheckFunc {
```
the param `rName` is not used

Lance52259 committed:
Update data_source_huaweicloud_iec_flavors_test.go

Change list: 
1. Remove useless function parameter rName.
2. Remove flavors name in config and remove parameter check.
3. Use HW_REGION_NAME to instead old type.
